pkg/server/plugin/nodeattestor/gcp/iit.go

```diff
@@ -94,6 +95,41 @@ func (p *IITAttestorPlugin) Attest(stream nodeattestor.Attest_PluginStream) erro
 	return nil
 }
 
+// ValidateAttestationAndExtractIdentityMetadata validates the attestation data, as well as the GCP identity
+// document that contains and returns the contained gcp.ComputeEngine object.
+func (p *BaseIITAttestorPlugin) ValidateAttestationAndExtractIdentityMetadata(stream nodeattestor.Attest_PluginStream, pluginName string) (gcp.ComputeEngine, error) {
```

Comment:
nit:  it might make sense for this to be a standalone exported function that takes the token "retriever" as a parameter. That would obviate the need for the Base plugin struct and subsequent embedding.